server/sonar-server/src/main/java/org/sonar/server/permission/index/AuthorizationDao.java

```diff
@@ -0,0 +1,218 @@
+    "      INNER JOIN users ON users.id = user_roles.user_id " +
+    "      WHERE " +
+    "        projects.qualifier = 'TRK' " +
+    "        AND projects.copy_component_uuid is NULL " +
+    "        {projectsCondition} " +
+    "      UNION " +
+
+    // groups without Anyone
+
+    "      SELECT " +
+    "      projects.uuid AS project, " +
+    "      projects.authorization_updated_at AS updated_at, " +
+    "      NULL  AS login, " +
+    "      groups.name  AS permission_group " +
+    "      FROM projects " +
+    "      INNER JOIN group_roles ON group_roles.resource_id = projects.id AND group_roles.role = 'user' " +
+    "      INNER JOIN groups ON groups.id = group_roles.group_id " +
+    "      WHERE " +
+    "        projects.qualifier = 'TRK' " +
+    "        AND projects.copy_component_uuid is NULL " +
+    "        {projectsCondition} " +
+    "        AND group_id IS NOT NULL " +
+    "      UNION " +
+
+    // Anyone groups
+
+    "      SELECT " +
+    "      projects.uuid AS project, " +
+    "      projects.authorization_updated_at AS updated_at, " +
+    "      NULL         AS login, " +
+    "      'Anyone'     AS permission_group " +
+    "      FROM projects " +
+    "      INNER JOIN group_roles ON group_roles.resource_id = projects.id AND group_roles.role='user' " +
+    "      WHERE " +
+    "        projects.qualifier = 'TRK' " +
+    "        AND projects.copy_component_uuid is NULL " +
+    "        {projectsCondition} " +
+    "        AND group_roles.group_id IS NULL " +
+    "    ) project_authorization";
+
+  List<Dto> selectAfterDate(DbClient dbClient, DbSession session, long afterDate) {
+    return doSelectAfterDate(dbClient, session, afterDate, Collections.emptyList());
+  }
+
+  List<Dto> selectAfterDate(DbClient dbClient, DbSession session, long afterDate, List<String> projectUuids) {
+    return executeLargeInputs(projectUuids, subProjectUuids -> doSelectAfterDate(dbClient, session, afterDate, subProjectUuids));
+  }
+
+  private List<Dto> doSelectAfterDate(DbClient dbClient, DbSession session, long afterDate, List<String> projectUuids) {
+    try {
+      Map<String, Dto> dtosByProjectUuid = Maps.newHashMap();
+      PreparedStatement stmt = null;
+      ResultSet rs = null;
+      try {
+        stmt = createStatement(dbClient, session, afterDate, projectUuids);
+        rs = stmt.executeQuery();
+        while (rs.next()) {
+          processRow(rs, dtosByProjectUuid);
+        }
+        return new ArrayList<>(dtosByProjectUuid.values());
+      } finally {
+        DbUtils.closeQuietly(rs);
+        DbUtils.closeQuietly(stmt);
+      }
+    } catch (SQLException e) {
+      throw new IllegalStateException("Fail to select authorizations after date: " + afterDate, e);
+    }
+  }
+
+  private static PreparedStatement createStatement(DbClient dbClient, DbSession session, long afterDate, List<String> projectUuids) throws SQLException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused method parameter "afterDate". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1172)